server/src/test/java/org/candlepin/model/ProductTest.java

```diff
@@ -160,6 +160,17 @@ public void testBaseEquality() {
         assertTrue(rhs.equals(lhs));
     }
 
+    /**
+     * This test case is disabled.
+     * We updated product equals method to check only UUIDs for derived product & provided products.
+     * This update is done to fix stackoverflow issues when derived or provided products create a cycle
+     * within the hierarchy.
+     * It avoids needing to traverse the object graph to ensure our children entities are part of the
+     * same organization, & ensures equality of two products is only "true" if both point to the same
+     * children. Without this, we risk linking across owners in cases where parallel creation of entities
+     * causes two otherwise identical products to exist at the same time for different orgs.
```

Comment:
Let's add this comment as part of the commit message (except the 'This test is disabled'), after the summary and empty line

Reply:
I would argue that we don't disable this test. We still want to verify that the equality works as expected with each field populated individually. We should add additional fields to the output of the source method to verify that derived and provided products are checked as expected.

Reply:
Modified the test case. Also added the summary in git commit.